Compiler.Lexer/Compiler.Lexer/Tokenizer.cs

```diff
@@ -0,0 +1,403 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Compiler.Lexer
+{
+    public partial class Tokenizer
+    {
+        private CharStream Stream;
+        private Token LastToken;
+
+        public Tokenizer(string source)
+        {
+            Stream = new CharStream(source);
+            LastToken = new Token
+                (TokenKind.UnknownToken, string.Empty, 0, 0);
```

Comment:
A class can be used here, such as TokenStream(CharStream).

It could also have a .Peek method. Shortening this class and having implementation in a class will be easier to understand and read.